dart-impl/base/src/logging.c

```diff
@@ -22,3 +22,29 @@ const int dart__base__unit_term_colors[DART_LOG_TCOL_NUM_CODES-1] = {
   DART_LOG_TCOL_RED,
   DART_LOG_TCOL_BLUE
 };
+
+
+enum dart__base__logging_loglevel
+dart__base__logging_env_loglevel()
+{
+  static enum dart__base__logging_loglevel level = DART_LOGLEVEL_TRACE;
+  static int log_level_parsed = 0;
+  if (!log_level_parsed) {
+    const char *envstr = getenv(DART_LOGLEVEL_ENVSTR);
+    if (envstr) {
```

Comment:
> DART should be considered a stand-alone deployable, I agree, so it should provide its own run-time config interface. 

Exactly that is my point: a user who wants to control the log verbosity of the underlying DART runtime would have to do so through API calls, either fiddling in an `ENV` or command line parser for flexibility himself or recompile his application. AFAICS, it is common practice to control the runtime through environment variables, e.g., OpenMP variables and MPI implementation-specific configuration flags.

> If at all, `getopt` should be used for this.
Not at all. `getopt` is used for _application_ specific command line parameters, not to control the underlying runtime. 

> The implementation of `dash::util::Config` is way more comprehensive than the current snippet in DART.
I agree that the parser is simple but it does what it needs to do. There is no fanciness required. There might be more DART-specific environment variables in the future, e.g., the number of threads, so we can build a central abstraction for `ENV` in DART. 

I strongly recommend keeping the DART runtime configuration flexible and independent from DASH. Logging is a good example: If  `DASH_ENABLE_LOGGING` was the only way of controlling log verbosity I can never suppress DASH log info and see full DART trace logs without manually issuing a DART API call.